File: app/src/main/java/com/moneycollect/example_java/activity/SettingsActivity.java
package com.moneycollect.example_java.activity;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.view.ViewGroup;
import android.view.Window;
import android.widget.ImageView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.preference.EditTextPreference;
import androidx.preference.PreferenceFragmentCompat;

import com.moneycollect.android.model.Address;
import com.moneycollect.android.model.request.RequestCreatePayment;
import com.moneycollect.android.model.response.PaymentMethod;
import com.moneycollect.example_java.R;
import com.moneycollect.example_java.databinding.SettingsActivityBinding;
import com.moneycollect.example_java.TestRequestData;

import java.math.BigInteger;
import java.util.ArrayList;


public class SettingsActivity extends AppCompatActivity {

    private SettingsActivityBinding viewBinding = null;

    @Override
    protected void onCreate(@Nullable @org.jetbrains.annotations.Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE);
        viewBinding = SettingsActivityBinding.inflate(getLayoutInflater());
        setContentView(viewBinding.getRoot());
        getWindow().setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT);
        getSupportFragmentManager()
                .beginTransaction()
                .replace(R.id.settingsContainer, new ConfigurationFragment())
                .commit();

        ImageView backIconIv = viewBinding.backIcon;
        backIconIv.setOnClickListener(v -> finish());
    }

    public static class ConfigurationFragment extends PreferenceFragmentCompat{

        @Override
        public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
            setPreferencesFromResource(R.xml.preferences, rootKey);

            EditTextPreference countryEtPreference= findPreference(getResources().getString(R.string.country_key));
            EditTextPreference currencyEtPreference= findPreference(getResources().getString(R.string.currency_key));
            EditTextPreference channelEtPreference= findPreference(getResources().getString(R.string.channel_key));


            countryEtPreference.setOnPreferenceChangeListener((preference, newValue) -> {
                changeCountry((String) newValue,countryEtPreference);
                return false;
            });


            currencyEtPreference.setOnPreferenceChangeListener((preference, newValue) -> {
                changeCurrency((String) newValue ,currencyEtPreference);
                return false;
            });

            channelEtPreference.setOnPreferenceChangeListener((preference, newValue) -> {
                changeChannel((String) newValue ,channelEtPreference);
                return false;
            });

            countryEtPreference.setText(TestRequestData.Companion.getCountry());
            currencyEtPreference.setText(TestRequestData.Companion.getCurrency());
            channelEtPreference.setText(TestRequestData.Companion.getFromChannel());
        }

        private void addPaymentMethod(String newValue, EditTextPreference addPMEtPreference) {
            PaymentMethod testAddPaymentMethod = new PaymentMethod(
                    "",
                    "",
                    "",
                    newValue,
                    null,
                    null
            );
            ArrayList<PaymentMethod> testLocalBankList = new ArrayList<>(TestRequestData.Companion.getTestLocalBankList());
            testLocalBankList.add(testAddPaymentMethod);
            TestRequestData.Companion.setTestLocalBankList(testLocalBankList);

            ArrayList<PaymentMethod> testAllBankList = new ArrayList<>(TestRequestData.Companion.getTestAllBankList());
            testAllBankList.add(testAddPaymentMethod);
            TestRequestData.Companion.setTestAllBankList(testAllBankList);
        }

        private void changeCountry(String newValue, EditTextPreference countryEtPreference) {
            countryEtPreference.setText(newValue);
            TestRequestData.Companion.setCountry(newValue);

            @SuppressLint("VisibleForTests")
            Address address= new Address(
                    TestRequestData.Companion.getAddress().getCity(),
                    newValue,
                    TestRequestData.Companion.getAddress().getLine1(),
                    TestRequestData.Companion.getAddress().getLine2(),
                    TestRequestData.Companion.getAddress().getPostalCode(),
                    TestRequestData.Companion.getAddress().getState()
                    );
            TestRequestData.Companion.setAddress(address);
            TestRequestData.Companion.getTestCustomer().address=address;
            TestRequestData.Companion.getTestCustomer().getShipping().address=address;
            TestRequestData.Companion.getTestBilling().address=address;
            TestRequestData.Companion.getTestRequestPayment().getShipping().address=address;
            TestRequestData.Companion.getTestConfirmPayment().getShipping().address=address;
        }

        private void changeCurrency(String newValue, EditTextPreference currencyEtPreference) {
            currencyEtPreference.setText(newValue);
            TestRequestData.Companion.setCurrency(newValue);
            TestRequestData.Companion.getTestRequestPayment().setCurrency(newValue);
            TestRequestData.Companion.getTestConfirmPayment().setCurrency(newValue);
        }

        private void changeAmount(String newValue, EditTextPreference amountEtPreference) {
            amountEtPreference.setText(newValue);
            TestRequestData.Companion.setAmount(new BigInteger(newValue));
            TestRequestData.Companion.getTestRequestPayment().setAmount(new BigInteger(newValue));
            TestRequestData.Companion.getTestConfirmPayment().setAmount(new BigInteger(newValue));
            for (RequestCreatePayment.LineItems item : TestRequestData.Companion.getLineItems()) {
                item.amount=new BigInteger(newValue);
            }
        }

        private void changeChannel(String newValue, EditTextPreference channelEtPreference) {
            channelEtPreference.setText(newValue);
            TestRequestData.Companion.setFromChannel(newValue);
            TestRequestData.Companion.getTestRequestPayment().setFromChannel(newValue);
        }
    }

}

File: app/src/main/java/com/moneycollect/example/activity/SettingsActivity.kt
package com.moneycollect.example.activity

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.View
import android.view.ViewGroup
import android.view.Window
import androidx.appcompat.app.AppCompatActivity
import androidx.preference.EditTextPreference
import androidx.preference.PreferenceFragmentCompat
import com.moneycollect.android.model.Address
import com.moneycollect.android.model.response.PaymentMethod
import com.moneycollect.example.R
import com.moneycollect.example.TestRequestData
import com.moneycollect.example.databinding.SettingsActivityBinding
import com.moneycollect.example.utils.CheckoutCreditCardCurrency
import java.math.BigInteger


class SettingsActivity : AppCompatActivity() {
    private var viewBinding: SettingsActivityBinding? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        supportRequestWindowFeature(Window.FEATURE_NO_TITLE)
        viewBinding = SettingsActivityBinding.inflate(layoutInflater)
        setContentView(viewBinding!!.root)
        window.setLayout(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.MATCH_PARENT)
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.settingsContainer, ConfigurationFragment())
            .commit()

        var backIconIv = viewBinding!!.backIcon
        backIconIv.setOnClickListener(View.OnClickListener {
            finish()
        })
    }

    class ConfigurationFragment : PreferenceFragmentCompat(){

        override fun onCreatePreferences(savedInstanceState: Bundle?, rootKey: String?) {
            setPreferencesFromResource(R.xml.preferences, rootKey)

            val countryEtPreference= findPreference<EditTextPreference>(resources.getString(R.string.country_key))
            val currencyEtPreference= findPreference<EditTextPreference>(resources.getString(R.string.currency_key))
//            val amountEtPreference= findPreference<EditTextPreference>(resources.getString(R.string.amount_key))
            val channelEtPreference= findPreference<EditTextPreference>(resources.getString(R.string.channel_key))

            val addPMEtPreference= findPreference<EditTextPreference>(resources.getString(R.string.add_local_payment_method_key))

            countryEtPreference?.setOnPreferenceChangeListener { preference, newValue ->
                changeCountry(newValue = newValue as String,countryEtPreference)
            }

            currencyEtPreference?.setOnPreferenceChangeListener { preference, newValue ->
                changeCurrency(newValue = newValue as String,currencyEtPreference)
            }

//            amountEtPreference?.setOnPreferenceChangeListener { preference, newValue ->
//                changeAmount(newValue = newValue,amountEtPreference)
//            }

            channelEtPreference?.setOnPreferenceChangeListener { preference, newValue ->
                changeChannel(newValue = newValue as String,channelEtPreference)
            }

//            addPMEtPreference?.setOnPreferenceChangeListener { preference, newValue ->
//                addPaymentMethod(newValue = newValue as String,addPMEtPreference)
//            }

            countryEtPreference?.text=TestRequestData.country
            currencyEtPreference?.text=TestRequestData.currency
//            amountEtPreference?.text=TestRequestData.amount.toString()
            channelEtPreference?.text=TestRequestData.fromChannel
        }

        private fun addPaymentMethod(newValue: String, addPMEtPreference: EditTextPreference?):Boolean{
            var  testAddPaymentMethod = PaymentMethod(
                "",
                "",
                "",
                newValue,
                null,
                null
            )
            var testLocalBankList=arrayListOf<PaymentMethod>()
            testLocalBankList.addAll(TestRequestData.testLocalBankList)
            testLocalBankList.add(testAddPaymentMethod)
            TestRequestData.testLocalBankList=testLocalBankList

            var testAllBankList= arrayListOf<PaymentMethod>()
            testAllBankList.addAll(TestRequestData.testAllBankList)
            testAllBankList.add(testAddPaymentMethod)
            TestRequestData.testAllBankList=testAllBankList
            return false
        }

        @SuppressLint("VisibleForTests")
        private fun changeCountry(newValue: String, countryEtPreference: EditTextPreference):Boolean{
            countryEtPreference.text=newValue
            TestRequestData.country = newValue
            var address= Address(
                line1 = TestRequestData.address.line1,
                line2 = TestRequestData.address.line2,
                city = TestRequestData.address.city,
                state = TestRequestData.address.state,
                postalCode = TestRequestData.address.postalCode,
                country = newValue,
            )
            TestRequestData.address=address
            TestRequestData.testCustomer.address=address
            TestRequestData.testCustomer.shipping?.address=address
            TestRequestData.testBilling.address=address
            TestRequestData.testRequestPayment.shipping?.address=address
            TestRequestData.testConfirmPayment.shipping?.address=address
            return false
        }

        private fun changeCurrency(newValue: String, currencyEtPreference: EditTextPreference):Boolean{
            currencyEtPreference.text=newValue
            TestRequestData.currency = newValue
            TestRequestData.testRequestPayment.currency = newValue
            TestRequestData.testConfirmPayment.currency = newValue
            return false
        }

        private fun changeAmount(newValue: Any, amountEtPreference: EditTextPreference):Boolean{
            amountEtPreference.text=newValue.toString()
            TestRequestData.amount = BigInteger(newValue.toString())
            TestRequestData.testRequestPayment.amount = BigInteger(newValue.toString())
            TestRequestData.testConfirmPayment.amount = BigInteger(newValue.toString())
            for (item in TestRequestData.lineItems) {
                item.amount=BigInteger(newValue.toString())
            }
            return false
        }

        private fun changeChannel(newValue: String, channelEtPreference: EditTextPreference):Boolean{
            channelEtPreference.text=newValue
            TestRequestData.fromChannel = newValue
            TestRequestData.testRequestPayment.fromChannel = newValue
            return false
        }
    }
}